js/models/lockingModel.js

```diff
@@ -1,33 +1,55 @@
 export default class LockingModel extends Backbone.Model {
 
-  set(attrName, attrVal, options = {}) {
-    const stopProcessing = (typeof attrName === 'object' || typeof attrVal !== 'boolean' || !this.isLocking(attrName));
-    if (stopProcessing) return super.set(...arguments);
-
-    const isSettingValueForSpecificPlugin = options?.pluginName;
-    if (!isSettingValueForSpecificPlugin) {
-      console.error('Must supply a pluginName to change a locked attribute');
-      options.pluginName = 'compatibility';
+  set(...args) {
+    if (typeof args[0] !== 'object') {
+      const [name, value] = args.splice(0, 2);
+      args.unshift({ [name]: value });
     }
-
-    const pluginName = options.pluginName;
-    if (this.defaults[attrName] !== undefined) {
-      this._lockedAttributes[attrName] = !this.defaults[attrName];
-    }
-    const lockingValue = this._lockedAttributes[attrName];
-    const isAttemptingToLock = (lockingValue === attrVal);
-
-    if (isAttemptingToLock) {
-      this.setLockState(attrName, true, { pluginName: pluginName, skipcheck: true });
-      return super.set(attrName, lockingValue);
+    const options = args[1] ?? {};
+    const attrValues = args[0];
+    const newValues = {};
+    for (const attrName in attrValues) {
+      const attrVal = attrValues[attrName];
+      const willChange = (this.attributes[attrName] !== attrValues[attrName]);
+      if (!willChange) continue;
+
+      const isNotLocking = (typeof attrVal !== 'boolean' || !this.isLocking(attrName));
+      if (isNotLocking) {
+        newValues[attrName] = attrVal;
+        continue;
+      }
+
+      const isSettingValueForSpecificPlugin = options?.pluginName;
+      if (!isSettingValueForSpecificPlugin) {
+        console.error('Must supply a pluginName to change a locked attribute');
+        options.pluginName = 'compatibility';
+      }
+
+      const pluginName = options.pluginName;
+      if (this.defaults[attrName] !== undefined) {
```

Comment:
I feel like I have missed something, but `defaults` are generally provided via a function so presumably array notation will not work as intended?

Reply:
I agree. Will change. Ta.